### ISSUE TITLE
Suggestion: Gamepad Index

### ISSUE THREAD

**Issue by SolarLune** [GoranM/bdx#468]
Hey, there. 

It'd be nice to have a gamepad index option to allow for specifying which particular gamepad the controls should map to. This would be useful for multiplayer, of course.

The only thing that makes this a bit more difficult is that input maps also need to be adjusted for multiple gamepads to work correctly. Maybe we could adopt an optional "g:CONTROL NAME:GAMEPAD INDEX" setup, where if there is a third argument, it would be the number of the gamepad index? So you'd have:

```java
Bdx.imaps.put("left", "k:left", "g:left"); // Left on the d-pad on the first gamepad, or perhaps on any gamepad?

Bdx.imaps.put("left", "k:left", "g:left:1");  // Left on the d-pad on the second gamepad
```

**Comment by GoranM** [comment]
Looking at the changes that need to be made, in order to support multiple gamepads, adjusting input maps doesn't strike me as the difficult part ... Anyway:

The input device is designated by the first marker, so I think it would be best to incorporate the device index into that marker, to have something like "g1:left".

"g" should just be an alias for "g0".

**Comment by SolarLune** [comment]
^ Huh, so you mean that "the hard part" is elsewhere? Perhaps in the Gamepad class itself?

The reason that I thought the gamepad index should come last is because the string is already delimited and split by colons, so it'd be relatively trivial to add the index at the end. But I guess it's easy to go with the gamepad index right after the first letter.

**Comment by GoranM** [comment]
Well, there are several places where changes would need to be made, in order to have a `Bdx.gamepads` list (Bdx, Gamepad, GdxProcessor, and maybe a few others), so that seems like the bulk of the work to me - InputMaps is just mapping to that, so that shouldn't be too hard.

The ":" delimiter is used to separate the device from the input marker ("device:input"). If you have an additional gamepad, you have an additional device, so the device marker should be different.

That's why "g1:left" makes more sense to me than "g:left:1".

**Comment by SolarLune** [comment]
So I'm taking a crack at this, but it's not really working - the gamepads aren't getting updated, though I can see they are being detected. Any ideas? I'm pretty sure I'm just forgetting to update all of the gamepads somewhere somehow, but input handling with LibGDX seems pretty complex (to me, anyway).

**Comment by rafcolson** [comment]
Can't help. But as I recall I always had to attach the gamepad before starting the game, else it was not recognized. (same with bge)

**Comment by GoranM** [comment]
I can't help you test this fully, because I only have one gamepad, but my one gamepad seems to work fine:

```java
    public void main(){
        Gamepad gp = Bdx.gamepads.get(0);
        if (gp.btnHit("x"))
            applyForce(0, 0, 300);
    }
```

Sacky jumps when I press x, so I assume it's dead only when multiple gamepads are connected?

Are events being delivered to handler methods on GamepadAdapter? Put some print lines in there to confirm that.

**Comment by SolarLune** [comment]
OK, I'm not sure - maybe I missed something yesterday, but it's working fine now. Tested with a Logitech F310 and a SNES USB controller.

PR submitted.

**Comment by SolarLune** [comment]
Whoops, I forgot to actually submit the PR. It's submitted now.

**Comment by SolarLune** [comment]
PR merged, so I'll close this now. Thanks!

**Comment by SolarLune** [comment]
Whoops, forgot to close this, haha. Closed!
